apps/full-site-editing/full-site-editing-plugin/full-site-editing/plugins/close-button-override/index.js

```diff
@@ -4,28 +4,58 @@
  * External dependencies
  */
 import domReady from '@wordpress/dom-ready';
+import ReactDOM from 'react-dom';
+import { __ } from '@wordpress/i18n';
+import { Button, Dashicon } from '@wordpress/components';
+import './style.scss';
 
 domReady( () => {
 	const { closeButtonLabel, closeButtonUrl, editorPostType } = fullSiteEditing;
 
+	// Only alter for the page and template part editors.
+	if ( 'wp_template_part' !== editorPostType && 'page' !== editorPostType ) {
+		return;
+	}
+
 	const editPostHeaderInception = setInterval( () => {
-		const closeButton = document.querySelector( '.edit-post-fullscreen-mode-close__toolbar a' );
+		// Cycle through interval until header toolbar is found.
+		const toolbar = document.querySelector( '.edit-post-header__toolbar' );
 
-		if ( ! closeButton ) {
+		if ( ! toolbar ) {
 			return;
 		}
 		clearInterval( editPostHeaderInception );
 
+		// Add components toolbar with override class name (original will be hidden in ./style.scss).
+		const componentsToolbar = document.createElement( 'div' );
+		componentsToolbar.className =
+			'components-toolbar edit-post-fullscreen-mode-close__toolbar edit-post-fullscreen-mode-close__toolbar__override';
+		toolbar.prepend( componentsToolbar );
+
+		// Create custom close button and append to components toolbar.
+		const newCloseButton = document.createElement( 'a' );
 		// When closing Template CPT (e.g. header) to navigate back to parent page.
 		if ( 'wp_template_part' === editorPostType && closeButtonUrl ) {
-			const newCloseButton = document.createElement( 'a' );
 			newCloseButton.href = closeButtonUrl;
 			newCloseButton.innerHTML = closeButtonLabel;
-			newCloseButton.className = 'components-button components-icon-button is-button is-default';
 			newCloseButton.setAttribute( 'aria-label', closeButtonLabel );
+			newCloseButton.className = 'components-button components-icon-button is-button is-default';
+		} else if ( 'page' === editorPostType ) {
+			newCloseButton.href = 'edit.php?post_type=page';
+			const newLabel = __( 'Pages' );
+			newCloseButton.innerHTML = newLabel;
+			newCloseButton.setAttribute( 'aria-label', newLabel );
+			newCloseButton.className = 'components-button';
 
-			const parentContainer = document.querySelector( '.edit-post-fullscreen-mode-close__toolbar' );
-			parentContainer.replaceChild( newCloseButton, closeButton );
+			ReactDOM.render(
+				<Button className="components-button components-icon-button">
```

Comment:
👍 I think it's a great idea to render the actual WordPress components to get a consistent design here :)

Reply:
> Personally, though, making the back button consistent for the template part editor makes sense to me. (but could use a plus one on that from a designer)

I agree.  I have currently updated the PR for this, and it looks like:

![Screen Shot 2019-11-04 at 1 43 34 PM](https://user-images.githubusercontent.com/28742426/68148126-62ddc380-ff09-11e9-9fb3-69b410977a07.png)

I have not touched anything for the posts editor yet though, but that would easily be something we could add to this if it is desired.

Reply:
nice. Let's wait for more design feedback, but this is looking really good!